homeassistant/components/insteon/switch.py

```diff
@@ -10,15 +10,15 @@
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up the Insteon switches from a config entry."""
 
-    def add_entities(discovery_info=None):
+    async def async_add_insteon_switch_entities(discovery_info=None):
```

Comment:
Keep it without `async` and instead annotate it with `@callback` for the dispatcher to run it in the event loop.

Reply:
`from homeassistant.core import callback`